### ISSUE TITLE
Multi-pass loss function

### ISSUE THREAD

**Issue by hawescm1** [vlfeat/matconvnet#1040]
I am trying to implement the pi-ensemble model from the paper:

Laine, S. and T. Aila. "Temporal Ensembling for Semi-Supervised Learning," arXiv:1610.02242, March 2017.

The pi-model requires you to send the same mini-batch of images through your network for two forward passes. On each forward pass different data augmentations and dropout are used. The resulting softmax outputs from the two forward passes are then differenced to form an unsupervised loss that is combined with standard cross-entropy loss and back propagated through the network to update the network weights.

How would I go about creating such a loss layer that spans two different forward passes?

I could replicate the whole network, making sure that I share the weights, and then inject my two different versions of the same mini-batch of images through each side of the network - then I can see how to create the loss layer that I would need. But this seems like an ugly implementation.

Is there a better way?
